version: '3.8'

services:
  postgres:
    image: postgres:16-alpine
    container_name: learning-nest-next-db-dev
    environment:
      POSTGRES_DB: learning_nest_next
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    ports:
      - '5432:5432'
    volumes:
      - postgres_dev_data:/var/lib/postgresql/data
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 10s
      timeout: 5s
      retries: 5

  api:
    build:
      context: .
      dockerfile: ./apps/api/Dockerfile
    container_name: learning-nest-next-api-dev
    depends_on:
      postgres:
        condition: service_healthy
    environment:
      DATABASE_HOST: postgres
      DATABASE_PORT: 5432
      DATABASE_USERNAME: postgres
      DATABASE_PASSWORD: postgres
      DATABASE_NAME: learning_nest_next
      NODE_ENV: development
      PORT: 4000
      FRONTEND_URL: http://localhost:3000
      JWT_SECRET: devsecret
    ports:
      - '4000:4000'
    restart: unless-stopped

  web:
    build:
      context: .
      dockerfile: ./apps/web/Dockerfile
    container_name: learning-nest-next-web-dev
    depends_on:
      - api
    environment:
      NEXT_TELEMETRY_DISABLED: '1'
    ports:
      - '3000:3000'
    restart: unless-stopped

volumes:
  postgres_dev_data:
